File: 0203-remove-linked-list-elements/0203-remove-linked-list-elements.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeElements(ListNode head, int val) {
        ListNode d = new ListNode(-1);
        ListNode ls = d;
        
        ListNode temp = head;
        while(temp != null){
            if(temp.val != val){
                ls.next = temp;
                ls = temp;
            }
            
            temp = temp.next;
        }
        
        // Assigning the last node to null
        if(ls != null){
            ls.next = null;
        }
        
        return d.next;
    }
}

File: 0203-remove-linked-list-elements/0203-remove-linked-list-elements.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} val
 * @return {ListNode}
 */
var removeElements = function(head, val) {
    let d = new ListNode(-1);
    let ls = d;
    
    let temp = head;
    while(temp){
        if(temp.val != val){
            ls.next = temp;
            ls = temp;
        }
        temp = temp.next;
    }
    
    // Assigning the Last node to null
    if(ls != null)
        ls.next = null;
    
    // Return the Head:
    return d.next;
};